relentless/optimize/objective.py

```diff
@@ -165,3 +168,66 @@ def design_variables(self):
         :class:`ObjectiveFunction` for which the result was constructed, mapped
         to the value of the variables at the time the result was constructed."""
         return self._design_variables
+
+class RelativeEntropy(ObjectiveFunction):
+    def __init__(self, target, simulation, potentials, thermo, dr, communicator=None):
+        if target.V is None or target.N is None:
+            raise ValueError('The target ensemble must have both V and N set.')
+
+        self.target = target
+        self.simulation = simulation
+        self.potentials = potentials
+        self.thermo = thermo
+        self.dr = dr
+        self.communicator = communicator
+
+    def compute(self, directory=None):
+        sim = self.simulation.run(self.target, self.potentials, directory, self.communicator)
+        sim_ens = self.thermo.extract_ensemble(sim)
+
+        # compute the relative entropy gradient by integration
+        g_tgt = self.target.rdf
+        g_sim = sim_ens.rdf
+        dvars = self.design_variables()
+        gradient = _collections.KeyedArray(keys=dvars)
+
+        for var in dvars:
+            update = 0
+            for i,j in self.target.pairs:
+                # evaluate derivative wrt design variable and interpolate with r
+                rs = self.potentials.pair.r
+                dus = self.potentials.pair.derivative((i,j),var)
+                for k,duk in enumerate(dus):
+                    if np.isinf(duk):
+                        rs = np.delete(rs,k)
+                        dus = np.delete(dus,k)
+                        break
+                dudvar = _math.Interpolator(rs,dus)
+
+                # find common domain to compare rdfs
+                r0 = max(g_sim[i,j].table[0][0],g_tgt[i,j].table[0][0])
+                r1 = min(g_sim[i,j].table[-1][0],g_tgt[i,j].table[-1][0],self.potentials.pair.rmax)
+                r = np.arange(r0,r1+0.5*self.dr,self.dr)
```

Comment:
@zeesherman do we really need to use a separate array of rs, or could we use the ones we used to tabulate the pair potential (cut down to the same bounds as the rdfs)? It seems like this would work just fine. Then we could get rid of dr as an argument here.

Reply:
We do still need to discuss this point. I am in favor of simplifying the number of parameters (i.e., removing `dr` if it is not really needed).